src/main/java/org/verdictdb/coordinator/AsyncAggExecutionRulerByConverge.java

```diff
@@ -0,0 +1,126 @@
+package org.verdictdb.coordinator;
+
+import org.verdictdb.VerdictResultStream;
+import org.verdictdb.VerdictSingleResult;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class AsyncAggExecutionRulerByConverge extends AsyncAggExecutionRuler {
+
+  VerdictResultStream resultStream;
+
+  Coordinator runningCoordinator;
+
+  // the values of the result should be within [(1-valueError), (1+valueError)] of the previous result.
+  // Otherwise, it is not converged.
+  Double valueError = 0.05;
+
+  // the #column of the result should be within [(1-columnError), (1+columnError)] of the previous result.
+  // Otherwise, it is not converged.
+  Double columnError = 0.05;
+
+  // key is the values of non-aggregated column, value is the values of aggregated column
+  HashMap<List<Object>, List<Object>> aggregatedMap = new HashMap<>();
+
+  AsyncAggExecutionRulerByConverge(VerdictResultStream resultStream, Coordinator runningCoordinator) {
+    this.resultStream = resultStream;
+    this.runningCoordinator = runningCoordinator;
+  }
+
+  public void setValueError(Double valueError) {
+    this.valueError = valueError;
+  }
+
+  public void setColumnError(Double columnError) {
+    this.columnError = columnError;
+  }
+
+  /**
+   * fetch the answer from stream until converge
+   * @return the accurate answer
+   */
+  @Override
+  public VerdictSingleResult fetchAnswerUntilAccurate() {
```

Comment:
Instead of letting this class directly fetching the answers, let it simply compute if the last result if good based on the collected answers. That is,

```java
// in ExecutionContext or SelectCoordinator
QueryResultAccuracyEstimator accEst = new THISCLASS();

while (resultStream.hasNext()) {
  VerdictSingleResult rs = resultStream.next();

  accEst.add(rs);
  if (accEst.isLastResultAccurate()) {
    return rs; // also do abort the query execution
  }
}
```